### PR TITLE
EG-1329, ValueMissingGroupError for DateInput

### PR DESCRIPTION
**Pull request by cmrhmrc**

### CODE REVIEW

File: app/models/ocelot/stanzas/RequiredErrorGroup.scala [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+package models.ocelot.stanzas
+
+import core.models.ocelot.stanzas.{ErrorCallout, VisualStanza, Populated}
+
+case class RequiredErrorGroup (override val next: Seq[String], group: Seq[ErrorCallout], stack: Boolean) extends VisualStanza with Populated
+
+object RequiredErrorGroup {
+  def apply(group: Seq[ErrorCallout]): RequiredErrorGroup =
+    RequiredErrorGroup(group.reverse.headOption.fold[Seq[String]](Seq.empty)(_.next), group, group.headOption.fold[Boolean](false)(_.stack))
```
Comment (woollett-r) [inline app/models/ocelot/stanzas/RequiredErrorGroup.scala:25]: Could you use lastOption in place of reverse.headOption?

File: conf/messages.cy [hunk 2]
```diff
@@ -14,9 +14,9 @@ getpagehelp.linkText = Help gyda’r dudalen hon (yn agor tab newydd)
 
 site.continue = Yn eich blaen
 
-label.day=Day
-label.month=Month
-label.year=Year
+label.day=Dydd
+label.month=Mis
+label.year=Blwyddyn
```
Comment (woollett-r) [inline conf/messages.cy]: I had "Blwydd". We will need to check later.

File: test/services/AggregatorSpec.scala [hunk 3]
```diff
@@ -855,4 +855,139 @@ class AggregatorSpec extends BaseSpec {
...
+      val phrase3 = Phrase(Vector("Error {0} {1}", "Welsh Error {0} {1}"))
+
+      val callout1 = ErrorCallout(phrase1, Seq(""), false)
+      val callout2 = ErrorCallout(phrase2, Seq(""), true)
+      val callout3 = ErrorCallout(phrase3, Seq(""), true)
+      val callout4 = ErrorCallout(phrase3, Seq(""), true)
+
+    }
+
+    "not create ImportantGroup  of length 1 when encountering isolated Important co" in new ErrorCalloutTest {
```
Comment (woollett-r) [inline test/services/AggregatorSpec.scala:872]: RequiredErrorGroup rather than ImportantGroup

File: test/services/AggregatorSpec.scala [hunk 4]
```diff
@@ -855,4 +855,139 @@ class AggregatorSpec extends BaseSpec {
...
+          callout,
+          callout1,
+          instruction)
+
+      val aggregatedStanzas: Seq[Stanza] = Aggregator.aggregateStanzas(Nil)(stanzas)
+      aggregatedStanzas(0) shouldBe callout
+      aggregatedStanzas(1) shouldBe callout1
+    }
+
+    "leave two Important callouts with stack set to false" in new ErrorCalloutTest {
```
Comment (woollett-r) [inline test/services/AggregatorSpec.scala:884]: Check test descriptions

File: app/models/ocelot/stanzas/RequiredErrorGroup.scala (hunk 1 above)
Comment (cmrhmrc) [reply app/models/ocelot/stanzas/RequiredErrorGroup.scala:25]: yes, updated

File: conf/messages.cy (hunk 2 above)
Comment (cmrhmrc) [reply conf/messages.cy]: I've updated this branch to the translation you've used to remove conflict

File: test/services/AggregatorSpec.scala (hunk 3 above)
Comment (cmrhmrc) [reply test/services/AggregatorSpec.scala:872]: done

File: test/services/AggregatorSpec.scala (hunk 4 above)
Comment (cmrhmrc) [reply test/services/AggregatorSpec.scala:884]: done

**Pull request merged**